### PR TITLE
Use enum for error messages IDs.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
@smarter check this out, @felixmulder told me that you had some opinions on the IDs.

File: compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala [hunk 1]
```diff
@@ -94,8 +95,8 @@ object messages {
 
 
   // Syntax Errors ---------------------------------------------------------- //
-  abstract class EmptyCatchOrFinallyBlock(tryBody: untpd.Tree, errNo: Int)(implicit ctx: Context)
-  extends Message(errNo) {
+  abstract class EmptyCatchOrFinallyBlock(tryBody: untpd.Tree, errNo: ErrorMessageID)(implicit ctx: Context)
+      extends Message(EmptyCatchOrFinallyBlockID) {
```
Comment (felixmulder) [inline compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala]: weid 4 space indentation here - let's get rid of that yes 😉

**Comment by ckipp01** [comment]
5 years later.... just looking for a bit of clarity. Why are `LazyErrorId` and  `NoExplanationId` negatives?

**Comment by nicolasstucki** [comment]
@ckipp01 it came from https://github.com/lampepfl/dotty/commit/e42bb303231b9c91f679c410d51455ba91df024d#diff-e49722a882822c3e89863359a84e3431cc09a254b44dbed7d0914c75576003a9R93. Unfortunately this is not really useful.

`NoExplanation` was created here https://github.com/lampepfl/dotty/commit/e24289ac19a21c6b3794d02e8fe42766224f173c. Not sure if that helps either.

**Pull request merged**